packages/provider/src/TiptapCollabProvider.ts

```diff
@@ -297,12 +308,22 @@ export class TiptapCollabProvider extends HocuspocusProvider {
 
     // if the first comment of a thread is deleted we also
     // delete the thread itself as the source comment is gone
-    if (commentIndex === 0) {
+    if (commentIndex === 0 && deleteThread) {
       this.deleteThread(threadId)
       return
     }
 
     if (commentIndex > 0) {
+      const comment = thread.get('comments').get(commentIndex)
+      const newComment = new Y.Map()
+
+      newComment.set('id', comment.get('id'))
+      newComment.set('createdAt', comment.get('createdAt'))
+      newComment.set('updatedAt', (new Date()).toISOString())
```

Comment:
`updatedAt` is specified as a number type, but put here as a string? Which is actually stored?

Reply:
Just replicated what we did in the `updateComment` function already. See https://github.com/ueberdosis/hocuspocus/blob/b973449d512d081073c9c0c3583aeb0fa7fd652c/packages/provider/src/TiptapCollabProvider.ts#L279

@janthurau may shine some light on this? The types say that resolvedAt is using an ISOString while createdAt and updatedAt are expected to be numbers - yet we saved ISOStrings to updatedAt before.

Reply:
strings usually are not preferably in yjs because they are mergable so if two users concurrently set the value it could just concatenate them, whereas a number it would just deterministically pick one of them

Reply:
I updated the typings for the createdAt and updatedAt values on the TCollabComment type to represent the ISO strings